### PR TITLE
Feat: added custom bit-depth support

### LINKED ISSUE #10: Generalize wave generators to handle custom bit-depth
**ODHack**
---
- Go to the [Nicera Labs OD Telegram](https://t.me/niceralabs)
- Present yourself and tell us which issue you want to pick
- Explain shortly how you want to tackle the issue
- Ask a question if you need some clarity
- When you're selected you will be assigned to the issue

**Description**
---
Generalize all wave generators in `utils.cairo` with a custom bit_depth parameter so that they handle a custom bit-depth.
Must handle at least: 4-bit, 8-bit (current), 16-bit, 24-bit

[Wiki Resource on bit-depth](https://en.wikipedia.org/wiki/Audio_bit_depth)

**Additional Information**
---------------------------
[Medium Difficulty  - Cairo - Medium Priority]

**Acceptance Criteria**
---

- [ ] Implement custom bit-depth 
- [ ] Write and pass unit tests
- [ ] Generate example files

### PR DESCRIPTION
**Pull request by silent-cipher**
Fixes https://github.com/cienicera/cairo-wave/issues/10

Added support for custom bit depth of 4-bit, 8-bit, 16-bit, 24-bit and 32-bit.

### CODE REVIEW

File: src/utils.cairo [hunk 1]
```diff
@@ -2,41 +2,67 @@ use core::option::OptionTrait;
 use core::traits::TryInto;
 const PRECISION: u64 = 1_000_000;
 
+
+fn get_max_value(bit_depth: u16) -> u64 {
```
Comment (tekkac) [inline src/utils.cairo:6]: An array could have worked too.
I assume we don't need a get_min_value, can you confirm ?

File: src/wave.cairo [hunk 2]
```diff
@@ -41,10 +45,26 @@ impl WavToBytes of Into<WavFile, ByteArray> {
 
         // Append data
         let mut count = 0;
-        while self.data.len() - count > 0 {
-            bytes.append_byte(*self.data[count]);
-            count += 1;
-        };
+        if self.bits_per_sample == 4_u16 {
```
Comment (tekkac) [inline src/wave.cairo:48]: good catch, 4-bit is trickier

**Review by tekkac — APPROVED** [review]
Lgtm :notes:

File: src/utils.cairo (hunk 1 above)
Comment (silent-cipher) [reply src/utils.cairo:6]: There is no need for get_min_value as far as we aren't concerned with the amplitude.

**Pull request merged**